core/feed.go

```diff
@@ -793,11 +793,19 @@ func getCommentsPostTitles(ctx context.Context, db *sql.DB, comments []*Comment,
 	}
 
 	for _, post := range posts {
-		comment, ok := postToComment[post.ID]
-		if !ok {
-			return fmt.Errorf("populating comments' postTitle, comment not found of post id %v", post.ID)
+		if post.Title == "" {
+			return fmt.Errorf("populating comments' postTitle, could not find post title of post id %v", post.ID)
+		}
+
+		if postTitles[post.ID] == post.Title {
+			continue
 		}
-		comment.PostTitle = post.Title
+
+		postTitles[post.ID] = post.Title
+	}
+
+	for _, comment := range postToComment {
```

Comment:
If we get rid of the `postToComment` map, couldn't this just be `for _, comment := range comments {`?